File: front/src/app/core/service/post.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Comment } from '../payload/response/comment-response';
import { Post } from '../payload/response/post-response';

@Injectable({
  providedIn: 'root'
})
export class PostService {

  private url: string = 'http://localhost:9000/api/post';

  constructor(private http: HttpClient) { }

  getPosts(): Observable<Post[]> {
    return this.http.get<Post[]>(this.url);
  }

  getPost(id: number): Observable<Post> {
    return this.http.get<Post>(this.url + '/' + id);
  }

  getComments(postId: number): Observable<Comment[]> {
    return this.http.get<Comment[]>(this.url + '/' + postId + '/comments');
  }

  addComment(postId: number, content: string): Observable<Comment> {
    const body = {'content': content};
    return this.http.post<Comment>(this.url + '/' + postId + '/comments', body);
  }
}

File: back/src/main/java/com/openclassrooms/mddapi/service/PostService.java
package com.openclassrooms.mddapi.service;

import com.openclassrooms.mddapi.dto.request.CommentRequest;
import com.openclassrooms.mddapi.dto.response.CommentResponse;
import com.openclassrooms.mddapi.model.Post;

import java.util.List;

public interface PostService {
    List<Post> getPosts();

    Post getPost(Long id);

    List<CommentResponse> getComments(Long id);

    CommentResponse addComment(Long postId, CommentRequest commentRequest);
}
